department-of-veterans-affairs/vets-api
# [56048] Mockdata sync api

## Summary
Creates an api only available in staging that takes an ICN parameter, queriers MPI test, creates yml from response, and returns the yml in a JSON response

- Api key added to parameter store
- department-of-veterans-affairs/devops#12781
- department-of-veterans-affairs/vsp-infra-application-manifests#1729

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#56048

## Testing done

- Specs
- Tested methods inside staging rails console

## How to test:
The MPI test instance will only work on dev and staging but we can test the Mockdata::MPI::Find methods in the staging console.
- Get on a staging pod
- Start rails console: `bundle exec rails c`
- Get an ICN of a user from the test-users csv
- copy 🍝 contents of `mocked_authentication/mockdata/mpi/find.rb` file into the console
- Call `MockedAuthentication::Mockdata::MPI::Find.new(icn: icn).perform`

Route:
- You can test the route locally but it will return an error because it cant connect to MPI:
- Add api_key to `settings.local.yml`
  ```yml
  sign_in:
    mockdata_sync_api_key: {your_key}
  ```
- Make a GET request to `mocked_authentication/mpi/mockdata/:icn' with header => `Authorization: Bearer {your_key}`
- You will get a bad request error because we cant connect to MPI locally
- You can change the value of `yml` in `mockdata#show` to return something random to see a successful call 

## What areas of the site does it impact?
mock authentication

## Acceptance criteria


## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_
